File: cat_numbered_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def cat_numbered(file_path, config):
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    with open(file_path, "r", encoding="utf-8") as f:
        return "\n".join(f"{i}: {line.rstrip()}" for i, line in enumerate(f, 1))

File: ts-tools/cat_numbered_tool.ts

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function catNumbered(filePath: string, config: any): Promise<string> {
    return new Promise((resolve, reject) => {
        checkFilePermission(filePath, "read", config)
            .then(() => {
                const resolvedPath = path.resolve(filePath);
                fs.readFile(resolvedPath, 'utf-8', (err, data) => {
                    if (err) {
                        reject(err);
                    } else {
                        const lines = data.split('\n');
                        const numberedLines = lines.map((line, i) => `${i + 1}: ${line}`);
                        resolve(numberedLines.join('\n'));
                    }
                });
            })
            .catch(reject);
    });
}

